### PR TITLE
Don't use keybindings reserved for users

### PR DESCRIPTION
**Pull request by r0man**
I initially picked the keybinding prefix `C-c l <key>` for CIDER log mode not being aware that `C-c` is reserved for users. I now changed this to `C-x M-l <key>`.

-----------------

Before submitting the PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [x] The commits are consistent with our [contribution guidelines](../blob/master/.github/CONTRIBUTING.md)
- [-] You've added tests (if possible) to cover your change(s)
- [x] All tests are passing (`eldev test`)
- [x] All code passes the linter (`eldev lint`) which is based on [`elisp-lint`](https://github.com/gonewest818/elisp-lint) and includes
  - [byte-compilation](https://www.gnu.org/software/emacs/manual/html_node/elisp/Byte-Compilation.html), [`checkdoc`](https://www.gnu.org/software/emacs/manual/html_node/elisp/Tips.html), [check-declare](https://www.gnu.org/software/emacs/manual/html_node/elisp/Declaring-Functions.html), packaging metadata, indentation, and trailing whitespace checks.
- [x] You've updated the [changelog](../blob/master/CHANGELOG.md) (if adding/changing user-visible functionality)
- [x] You've updated the [user manual](../blob/master/doc) (if adding/changing user-visible functionality)

Thanks!

*If you're just starting out to hack on CIDER you might find this [section of its
manual][1] extremely useful.*

[1]: https://docs.cider.mx/cider/contributing/hacking.html

### CODE REVIEW

File: cider-log.el [hunk 1]
```diff
@@ -797,12 +797,12 @@ The KEYS are used to lookup the values and are joined by SEPARATOR."
 (defvar cider-log-mode-map
   (let ((map (make-sparse-keymap)))
     (set-keymap-parent map logview-mode-map)
-    (define-key map (kbd "C-c l a") #'cider-log-appender)
-    (define-key map (kbd "C-c l c") #'cider-log-consumer)
-    (define-key map (kbd "C-c l e") #'cider-log-event)
-    (define-key map (kbd "C-c l f") #'cider-log-framework)
-    (define-key map (kbd "C-c l i") #'cider-log-info)
-    (define-key map (kbd "C-c l l") #'cider-log)
+    (define-key map (kbd "C-x l a") #'cider-log-appender)
```
Comment (bbatsov) [inline cider-log.el:800]: We can't use `C-x l` directly for various reasons (e.g. a built-in command is using it and it's not a mode map prefix), it should be something like `C-c C-l` or `C-x C-l` (provided we're not already using them for something else).

Comment (r0man) [reply cider-log.el]: I changed it to `C-x C-l` now.

**Comment by bbatsov** [comment]
Damn, just saw that `C-x C-l` is bound to `downcase-region` in Emacs, so it's a no go as well. `C-c C-l` is `cider-load-file`. You'll have to get more creative. :-)

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 ### New features
 
+- [#3367](https://github.com/clojure-emacs/cider/pull/3367) Change CIDER Log Mode keybindings to use `C-c M-l <key>` instead of user reserved bindings.
```
Comment (bbatsov) [inline CHANGELOG.md:7]: That's not needed, as `cider-log-mode` hasn't been released yet. :-)

Comment (r0man) [reply CHANGELOG.md]: I removed it.

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**